hledger-lib/Hledger/Read/RulesReader.hs

```diff
@@ -728,15 +737,22 @@ maybeNegate _ origbool = origbool
 -- Note conditional blocks' patterns are matched against an approximation of the
 -- CSV record: all the field values, without enclosing quotes, comma-separated.
 --
-getEffectiveAssignment :: CsvRules -> CsvRecord -> HledgerFieldName -> Maybe FieldTemplate
-getEffectiveAssignment rules record f = lastMay $ map snd $ assignments
+getEffectiveAssignment
```

Comment:
I think a couple of comments & haddocks might need updating, eg lines 734 and 751.